vtm/src/org/oscim/layers/tile/vector/labeling/LabelLayer.java

```diff
@@ -24,22 +25,28 @@
 import org.oscim.layers.Layer;
 import org.oscim.layers.tile.MapTile;
 import org.oscim.layers.tile.TileManager;
+import org.oscim.layers.tile.ZoomLimiter;
 import org.oscim.layers.tile.vector.VectorTileLayer;
 import org.oscim.map.Map;
 import org.oscim.utils.async.SimpleWorker;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class LabelLayer extends Layer implements Map.UpdateListener, TileManager.Listener {
+public class LabelLayer extends Layer implements Map.UpdateListener, TileManager.Listener,
+        ZoomLimiter.IZoomLimiter {
 
     static final Logger log = LoggerFactory.getLogger(LabelLayer.class);
 
     static final String LABEL_DATA = LabelLayer.class.getName();
 
     private static final long MAX_RELABEL_DELAY = 100;
 
+    /* Set to maximum "zoom-min" value for labels in render themes */
+    public static final int ZOOM_LIMIT = 18;
```

Comment:
Of course have to make it `private` or remove `final` ;)

Reply:
`ZOOM_LIMIT` shouldn't be large or else rendering is similar like now. A sane default could be `17`.

Nevertheless a side effect is that if set in themes `zoom-min` >  `ZOOM_LIMIT` then the labels / symbols are not rendered!
We somehow need to not break themes functionality here.